otj-metrics-actuate/src/main/java/com/opentable/metrics/actuate/health/CodahaleHealthIndicatorConfiguration.java

```diff
@@ -26,25 +26,34 @@
 
 /**
  * Presents {@link HealthCheck} beans as a composite {@link HealthIndicator}.
+ * The purpose of this is to allow Actuator to see DropWizard health checks
  *
  */
 @Configuration
+// Only wires if drop-wizard is enabled? Can you elucidate further dmitry?
 @ConditionalOnEnabledHealthIndicator("drop-wizard")
 public class CodahaleHealthIndicatorConfiguration extends  CompositeHealthIndicatorConfiguration<CodahaleHealthIndicatorConfiguration.Adapter, HealthCheck> {
 
     private final Map<String, HealthCheck> dropWizardChecks;
 
+    // How does this inject work - what's the key?
+    // Should it be Optional<Map> ?
```

Comment:
@mikebell90 I found it here https://github.com/opentable/otj-metrics/blob/e59a495ba6e0c38f46330041a08c81f39304043c/otj-metrics-core/src/main/java/com/opentable/metrics/health/HealthConfiguration.java#L73 (also was surprised) Apparently this is as spring feature, similar to injecting lists. Key is the bean name. It'll double check behavior in case no such beans

Reply:
Cool!